Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
Yes, that's perfectly valid usage and `Returns` API allows it. But you cannot call it for void method, only `WhenDo` and `CallActions` work for them.
We will come up to the best solution I'm sure 😉 

PS I want to take a part in the NSub-AutoFixture development, please let me know.